File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_CSharp/Surface.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: Surface.cs
//
//--------------------------------------------------------------------------

using System;

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples
{
    class Surface
    {
        public Func<Vector, Color> Diffuse;
        public Func<Vector, Color> Specular;
        public Func<Vector, double> Reflect;
        public double Roughness;

        public Surface(Func<Vector, Color> Diffuse,
                        Func<Vector, Color> Specular,
                        Func<Vector, double> Reflect,
                        double Roughness)
        {
            this.Diffuse = Diffuse;
            this.Specular = Specular;
            this.Reflect = Reflect;
            this.Roughness = Roughness;
        }

    }
}

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_VisualBasic/Surface.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: Surface.vb
'
'--------------------------------------------------------------------------

Public Class Surface
    Public Diffuse As Func(Of Vector, Color)
    Public Specular As Func(Of Vector, Color)
    Public Reflect As Func(Of Vector, Double)
    Public Roughness As Double

    Public Sub New(ByVal diffuse As Func(Of Vector, Color), ByVal specular As Func(Of Vector, Color),
                   ByVal reflect As Func(Of Vector, Double), ByVal roughness As Double)
        Me.Diffuse = diffuse
        Me.Specular = specular
        Me.Reflect = reflect
        Me.Roughness = roughness
    End Sub
End Class